control/setup_control.go

```diff
@@ -0,0 +1,399 @@
+package control
+
+import (
+	"context"
+	"sync"
+
+	"go.uber.org/multierr"
+	"go.viam.com/utils"
+
+	"go.viam.com/rdk/logging"
+	rdkutils "go.viam.com/rdk/utils"
+)
+
+/*
+SetupControlLoop
+tunePID
+GetState ?
+SetState ?
+*/
+
+// rPiGain is 1/255 because the PWM signal on a pi (and most other boards)
+// is limited to 8 bits, or the range 0-255.
+const rPiGain = 0.00392157
+
+var (
+	// default derivative type is "backward1st1".
+	derivativeType   = "backward1st1"
+	loopFrequency    = 50.0
+	sumIndex         = 1
+	linearPIDIndex   = 2
+	angularPIDIndex  = -1
+	controllableType = "motor_name"
+)
+
+// PIDLoop is used for setting up a PID control loop.
+type PIDLoop struct {
+	BlockNames              map[string][]string
+	ControlConf             Config
+	ControlLoop             *Loop
+	Options                 Options
+	Controllable            Controllable
+	logger                  logging.Logger
+	activeBackgroundWorkers sync.WaitGroup
+}
+
+// PIDConfig is values needed to configure a PID control loop.
+type PIDConfig struct {
+	Type string  `json:"type,omitempty"`
+	P    float64 `json:"p"`
+	I    float64 `json:"i"`
+	D    float64 `json:"d"`
+}
+
+// Options contains values used for a control loop.
+type Options struct {
+	// PositionControlUsingTrapz
```

Comment:
[minor] can we add a quick description of each of these, and maybe when they are intended to be used for the optional ones?